packages/jest-jasmine2/src/__tests__/integration/lifecycle_hooks_test.js

```diff
@@ -56,3 +56,34 @@ describe('test lifecycle hooks', () => {
     it('does it 1', pushMessage('outer it 1'));
   });
 });
+
+describe('shares this context between nested describe blocks', () => {
+  const actions = [];
+  function pushMessage(message) {
+    actions.push(message);
+  }
+
+  beforeEach(function() {
+    this.x = 42;
+    pushMessage('beforeEach1-' + this.x);
+  });
+
+  describe('Something', () => {
+    beforeEach(function() {
+      pushMessage('beforeEach2-' + this.x);
+    });
+
+    it('contains all actions', function() {
+      this.x = 42;
```

Comment:
Yeah! Actually, it **should** be removed completely. As I'd like to test that `this.x` has the same value that was initially set in the first `beforeEach()` function!